### DISCUSSION TITLE
Removing component from within the global mount/create hook

### DISCUSSION THREAD

**Discussion by m-jojo-s** [GrapesJS/grapesjs#4135] [Q&A] [answered]
I want to check if a newly added component (an image in this case) can be added or not. I tried removing it and it does remove the model so it's no longer interactive but the view isn't removed. Would like it if it could be removed before rendering but attemping to remove it  from the 'component:create' hook had no effect.

```
editor.on('component:mount', (component) => {
        if (someCondition) {
            component.remove();
        }
});
```

**Comment by artf** [comment, marked as answer]
https://github.com/artf/grapesjs/issues/4136
